### PR TITLE
Add ubuntu 20 container image

### PR DESCRIPTION
**Pull request by VJanKraemer**
Adds the Ubuntu 20.04 container image and adapts the 18.04 image so both images are coherent in the build recipes!

### CODE REVIEW

File: containers/ci-ubuntu-20.04/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,41 @@
+FROM ubuntu:20.04
```
Comment (MariusBgm) [inline containers/ci-ubuntu-20.04/Dockerfile:1]: This is basically the same as the 18.04 Dockerfile with a different base ubuntu and gcc version, if I'm not mistaken.
Would it make sense to have a single, parameterized Dockerfile for ubuntu?
e.g.:
```suggestion
ARG UBUNTU_VERSION=18.04
FROM ubuntu:${UBUNTU_VERSION}

ARG GCC_VERSION=8

```

and later on `apt install ... gcc-${GCC_VERSION} g++-${GCC_VERSION}`

Comment (VJanKraemer) [reply containers/ci-ubuntu-20.04/Dockerfile]: Done

File: .github/workflows/build-sil-kit-ci-image.yaml [hunk 2]
```diff
@@ -19,19 +17,100 @@ jobs:
...
+  ubuntu-20.04:
+    id: ubuntu2004
+    name: Push Docker image to ghcr.io registry
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v4
+
+      - name: Log in to Docker Hub
+        uses: docker/login-action@f4ef78c080cd8ba55a85445d5b36e214a81df20a
```
Comment (MariusBgm) [inline .github/workflows/build-sil-kit-ci-image.yaml:52]:
```suggestion
        uses: docker/login-action@v2
```

File: .github/workflows/build-sil-kit-ci-image.yaml [hunk 3]
```diff
@@ -19,19 +17,100 @@ jobs:
...
+    runs-on: ubuntu-latest
+    env:
+      REGISTRY: ghcr.io
+      IMAGE_NAME: MariusBgm/sil-kit-ci-ubuntu-22.04
+    steps:
+      - name: Check out the repo
+        uses: actions/checkout@v4
+
+      - name: Log in to Docker Hub
+        uses: docker/login-action@f4ef78c080cd8ba55a85445d5b36e214a81df20a
```
Comment (MariusBgm) [inline .github/workflows/build-sil-kit-ci-image.yaml:91]:
```suggestion
        uses: docker/login-action@v2
```

**Pull request merged**